crates/router/src/connector/bluesnap.rs

```diff
@@ -80,25 +80,33 @@ impl ConnectorCommon for Bluesnap {
         res: Response,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
         logger::debug!(bluesnap_error_response=?res);
-        let response: bluesnap::BluesnapErrorResponse = res
+        let response: bluesnap::BluesnapErrors = res
             .response
             .parse_struct("BluesnapErrorResponse")
             .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
 
-        let response_error_message = response.message.first().map_or(
-            ErrorResponse {
+        let response_error_message = match response {
+            bluesnap::BluesnapErrors::PaymentError(error_res) => error_res.message.first().map_or(
+                ErrorResponse {
+                    status_code: res.status_code,
+                    code: consts::NO_ERROR_CODE.to_string(),
+                    message: consts::NO_ERROR_MESSAGE.to_string(),
+                    reason: None,
+                },
+                |error_response| ErrorResponse {
+                    status_code: res.status_code,
+                    code: error_response.code.clone(),
+                    message: error_response.description.clone(),
+                    reason: None,
+                },
+            ),
```

Comment:
In case of multiple error responses you can't ignore the rest of the errors and take up only the first one. You can take this up any other day to support multiple error responses.